backend/secrets/error.go

```diff
@@ -0,0 +1,34 @@
+package secrets
+
+import "fmt"
+
+type ErrInvalidSecretInfo string
+type ErrNoProviderDefined struct{}
```

Comment:
For errors with an invariant message, it's a little better and a little more idiomatic to use

```
var ErrNoProviderDefined = errors.New("no provider defined")
```

These can be easily compared by value, for example, `if err == ErrNoProviderDefined { // handle error }`.

In your use of ErrNoProviderDefined, you've returned a pointer value, which would not compare as easily. Similar for ErrSecretsNotSupported.

Reply:
Fixed.